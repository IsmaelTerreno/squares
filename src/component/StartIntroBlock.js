import React from 'react';
import PropTypes from 'prop-types';
import logo from '../images/app-main-logo.png';
import { Link } from 'react-router-dom';

/**
 * Intro block to show the first info and start action.
 */
const StartIntroBlock = ({title, description, buttonLegend, buttonUrlAction}) => (
    <div className="animated bounceIn">
      <div className="card" style={{padding: '0.5rem', marginTop:'8rem'}}>
        <img className="card-img-top" style={{width:'5rem', margin: '0 auto'}} src={logo} alt="Square" />
        <div className="card-body">
          <h5 className="card-title">{title}</h5>
          <p className="card-text">{description}</p>
          <div className="row">
            <div className="col">
              <Link role="button" to={buttonUrlAction} className="btn btn-primary animated rubberBand">Start to create</Link>
            </div>
            <div className="col">
              <Link role="button" to="/square/gallery" className="btn btn-primary animated rubberBand">Gallery</Link>
            </div>
          </div>
        </div>
      </div>
    </div>
);

StartIntroBlock.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    buttonLegend: PropTypes.string,
    buttonUrlAction: PropTypes.string,
};

export default StartIntroBlock;





