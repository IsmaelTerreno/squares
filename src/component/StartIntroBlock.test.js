import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";
import StartIntroBlock from './StartIntroBlock';

configure({ adapter: new Adapter() });

describe('StartIntroBlock', () => {

  test('Should renders appropriately', () => {
    const wrapper = shallow(
      <StartIntroBlock
        title="Art with Squares"
        description="Create customized stylish squares to have your own gallery and show them to the world."
        buttonLegend="Start to create"
        buttonUrlAction="/square"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
