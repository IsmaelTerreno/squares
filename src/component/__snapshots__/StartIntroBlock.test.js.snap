// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartIntroBlock Should renders appropriately 1`] = `
<div
  className="animated bounceIn"
>
  <div
    className="card"
    style={
      Object {
        "marginTop": "8rem",
        "padding": "0.5rem",
      }
    }
  >
    <img
      alt="Square"
      className="card-img-top"
      src="app-main-logo.png"
      style={
        Object {
          "margin": "0 auto",
          "width": "5rem",
        }
      }
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Art with Squares
      </h5>
      <p
        className="card-text"
      >
        Create customized stylish squares to have your own gallery and show them to the world.
      </p>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <Link
            className="btn btn-primary animated rubberBand"
            replace={false}
            role="button"
            to="/square"
          >
            Start to create
          </Link>
        </div>
        <div
          className="col"
        >
          <Link
            className="btn btn-primary animated rubberBand"
            replace={false}
            role="button"
            to="/square/gallery"
          >
            Gallery
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
